demos/python_demos/action_recognition/action_recognition.py

```diff
@@ -69,18 +68,12 @@ def build_argparser():
 
 def main():
     args = build_argparser().parse_args()
-
-    full_name = path.basename(args.input)
-    extension = path.splitext(full_name)[1]
-
-    if '.txt' in  extension:
-        with open(args.input) as f:
-            videos = [line.strip() for line in f.read().split('\n')]
-    else:
-        videos = [args.input]
-
-    if not args.input:
+    input_path = args.input
+    if not input_path:
         raise ValueError('--input option is expected')
+    if not path.isfile(input_path):
+        raise ValueError('The input file does not exists')
+    video = [input_path]
```

Comment:
We don't need any of this. VideoCapture fails when it encounters empty string. And non-file inputs are valid. For example `0` for camera.

So run_pipeline() could just take `args.input`.